src/librustc_resolve/lib.rs

```diff
@@ -1610,6 +1610,14 @@ impl<'a> Resolver<'a> {
         DefCollector::new(&mut definitions, Mark::root())
             .collect_root(crate_name, session.local_crate_disambiguator());
 
+        let mut extern_prelude: FxHashSet<Name> =
+            session.opts.externs.iter().map(|kv| Symbol::intern(kv.0)).collect();
+        if !attr::contains_name(&krate.attrs, "no_std") {
+            extern_prelude.insert(Symbol::intern("std"));
```

Comment:
This adds `std` to the prelude for `no_core` crates.
(The logic should mirror `fn maybe_inject_crates_ref` in `libsyntax/std_inject.rs`.)

Reply:
Additionally, should `core` be available in extern prelude by default (for not `no_std`/`no_core` crates)?
Also, should other stable crates from the standard distribution like `proc_macro` be available?
I'd say no for a start, they can always be added later.

Reply:
Fixed this to share the same logic as `maybe_inject_crates_ref`.